api/app/signals/apps/questionnaires/services/session.py

```diff
@@ -190,11 +190,7 @@ def get_extra_properties(self, category_url=None):
         # TBD, still how do we mark answers with the proper category
         extra_props = []
         for answer in self._path_answers_by_question_id.values():
-            entry = {
-                'id': answer.question.analysis_key,
-                'label': answer.question.short_label,
-                'answer': answer.payload,
-            }
+            entry = answer.extra_property()
```

Comment:
See comment about where the `extra_property` conversion functionality should live. My proposal would mean this line would be something like `entry = self.answer_service.convert_to_extra_property(answer)`.